### ISSUE TITLE
release new version

### ISSUE THREAD

**Issue by thkoch2001** [emacs-lsp/lsp-mode#454]
The last version (4.2) was committed in June.
I'd love to try to get lsp-mode into the next Debian.

**Comment by vibhavp** [comment]
I think we can release 5.0 once https://github.com/emacs-lsp/lsp-mode/pull/453 and https://github.com/emacs-lsp/lsp-mode/pull/448 have been merged.

**Comment by MaskRay** [comment]
#453 has been merged. #448 

Shall we also migrate away from `lsp-client-on-notification` and use `cl-defmethod` instead? e.g.

```elisp
(cl-defmethod eglot-handle-notification
  (server (_method (eql textDocument/publishDiagnostics)) &key uri diagnostics)
```

The clients I know that make use of this API is: emacs-ccls emacs-cquery lsp-java

**Comment by yyoncho** [comment]
Released 5.0.
